### ISSUE TITLE
[iOS] Accepting incoming call doesn't turn on microphone

### ISSUE THREAD

**Issue by maharjanaman** [QuickBlox/quickblox-react-native-sdk#76]
I'm trying to implement audio. Have enable all necessary background modes,
```
<key>UIBackgroundModes</key>
<array>
  <string>audio</string>
  <string>fetch</string>
  <string>remote-notification</string>
  <string>voip</string>
</array>
```
requested both, `PERMISSIONS.IOS.CAMERA, PERMISSIONS.IOS.MICROPHONE` permission, but when accepting incoming calls, the microphone turns on for instance, and goes back off.

However, when initiating a call from an iOS device the microphone stays ON. It's just when accepting incoming calls, the microphone gets turned OFF.

Here's the log,
[call-log.txt](https://github.com/QuickBlox/quickblox-react-native-sdk/files/14353924/call-log.txt)

**Comment by maharjanaman** [comment]
This seems to be issue from https://github.com/react-native-webrtc/react-native-incall-manager. Closing this for now.

**Comment by IlliaChemolosov** [comment]
@maharjanaman 
Hi there! 🎧 This post is all about Problems with Audio in React Native Calls on iOS 📞🔊

In the release of [QuickBlox React Native SDK 0.9.8](https://www.npmjs.com/package/quickblox-react-native-sdk), we've made some improvements to the call lifecycle event queue and audio session settings to enhance interaction with other libraries. However, there's still a chance of conflicts within the application code itself. To mitigate this, we recommend carefully considering the sequence of acquiring and releasing the audio session. Take a cue from our example with ringtones to prevent conflicts with audio below.  👇

Ringback Tones (tones heard by the operator while waiting for call confirmation or rejection):

🎵 Start playing such a ringback tone after calling the QB.webrtc.call(callParams) method in the SDK.
🛑 Stop playing the tone in the following cases:
  - When the opponent accepts the call (event QB.webrtc.EVENT_TYPE.ACCEPT).
  - When the opponent rejects the call (event QB.webrtc.EVENT_TYPE.REJECT).
  - When the opponent does not answer (event QB.webrtc.EVENT_TYPE.NOT_ANSWER).
  - When the initiator prematurely ends the call (before calling the QB.webrtc.hangUp() method).

Ringtone (audible signal heard during an incoming call):

🎶 Start playing such a ringtone after receiving the call signal (event QB.webrtc.EVENT_TYPE.CALL).
🛑 Stop playing the tone in the following cases:
  - When the call is accepted (before calling QB.webrtc.accept({ sessionId: session.id })).
  - When the call is rejected (before calling QB.webrtc.reject({ sessionId: session.id })).
  - When the call is not accepted within a limited time (event QB.webrtc.EVENT_TYPE.NOT_ANSWER).
  - When the opponent cancels the call (event QB.webrtc.EVENT_TYPE.HANG_UP).

We've also updated our example to align with these changes. QuickBlox React Native WebRTC Sample version 1.1.1 is available at: [GitHub](https://github.com/QuickBlox/quickblox-react-native-samples) 🚀
Release notes can be found in the package repository: [Release Notes](https://github.com/QuickBlox/quickblox-react-native-samples/releases/tag/webrtc-sample-v1.1.1) 📄👀

[Discord](https://discord.com/channels/1042743094833065985/1164589205456957450)
